### PR TITLE
Upgrade wabt to 1.0.31

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

File: internal/wasm/func_validation.go [hunk 1]
```diff
@@ -868,9 +868,19 @@ func (m *Module) validateFunctionWithMaxStackValues(
 			pc += num - 1
 		} else if op == OpcodeMiscPrefix {
 			pc++
-			// Miscellaneous instructions come with two bytes which starts with OpcodeMiscPrefix,
-			// and the second byte determines the actual instruction.
-			miscOpcode := body[pc]
+			// A misc opcode is encoded as an unsigned variable 32-bit integer.
+			miscOp32, num, err := leb128.LoadUint32(body[pc:])
```
Comment (mathetake) [inline internal/wasm/func_validation.go:872]: I didn't know that amazingly misc opcodes are encoded in leb128 U32...

File: internal/integration_test/spectest/spectest.go [hunk 2]
```diff
@@ -836,38 +836,3 @@ func requireStripCustomSections(t *testing.T, binary []byte) []byte {
 	}
 	return out.Bytes()
 }
-
-// TestBinaryEncoder ensures that binary.EncodeModule produces exactly the same binaries
-// for wasm.Module via binary.DecodeModule modulo custom sections for all the valid binaries in spectests.
-func TestBinaryEncoder(t *testing.T, testDataFS embed.FS, enabledFeatures api.CoreFeatures) {
```
Comment (mathetake) [inline internal/integration_test/spectest/spectest.go:842]: deleted this test which is used to develop our own fuzzer, but now it's gone. so I deleted this

**Comment by mathetake** [comment]
this comes with https://github.com/WebAssembly/wabt/issues/2022

**Comment by mathetake** [comment]
will update the spectest version in the separate PR

**Comment by mathetake** [comment]
feels like vagrant error is persistent :D

File: internal/wasm/func_validation.go (hunk 1 above)
Comment (codefromthecrypt) [reply internal/wasm/func_validation.go:872]: oh good that you found this, though scary to expect to need this encoding :D

**Review by codefromthecrypt — APPROVED** [review]
thanks!

**Pull request merged**